tests/testthat/test-replace_lower_right_with_NA.R

```diff
@@ -1,100 +1,222 @@
+  expected <- matrix(
+    c(1, 2, 3, NA),
+    nrow = 2,
+    byrow = TRUE
+  )
+  result <- replace_lower_right_with_NA(two_by_two)
+  expect_identical(result, expected)
+})
+
+test_that("replace_lower_right_with_NA handles matrix with existing NAs", {
+  na_matrix <- matrix(
     c(
-      1, 2, 3, 4,
-      5, 6, 7, NA,
-      9, 10, NA, NA,
-      13, NA, NA, NA
+      1, 2, NA,
+      4, 5, 6,
+      7, 8, 9
     ),
-    nrow = 4,
+    nrow = 3,
+    ncol = 3,
     byrow = TRUE
   )
-  expect_equal(replace_lower_right_with_NA(square_matrix),
-    expected_square,
-    tol = 1e-6
+  expected <- matrix(
+    c(
+      1, 2, NA,
+      4, 5, NA,
+      7, NA, NA
+    ),
+    nrow = 3,
+    byrow = TRUE
   )
+  result <- replace_lower_right_with_NA(na_matrix)
+  expect_identical(result, expected)
+})
 
-  # Test case 2: Rectangular matrix (more rows than columns)
-  rect_matrix1 <- matrix(1:20, nrow = 5, ncol = 4, byrow = TRUE)
-  expected_rect1 <- matrix(c(
-    1, 2, 3, 4,
-    5, 6, 7, 8,
-    9, 10, 11, NA,
-    13, 14, NA, NA,
-    17, NA, NA, NA
-  ), nrow = 5, byrow = TRUE)
-  expect_equal(replace_lower_right_with_NA(rect_matrix1),
-    expected_rect1,
-    tol = 1e-6
+test_that("replace_lower_right_with_NA handles one-row matrix", {
+  one_row <- matrix(1:5, nrow = 1)
+  expected <- matrix(
+    c(1, NA, NA, NA, NA),
+    nrow = 1
   )
+  result <- replace_lower_right_with_NA(one_row)
+  expect_identical(result, expected)
+})
 
-  # Test case 3: Rectangular matrix (more columns than rows)
-  rect_matrix2 <- matrix(1:20, nrow = 4, ncol = 5, byrow = TRUE)
-  expected_rect2 <- matrix(c(
-    1, 2, 3, 4, NA,
-    6, 7, 8, NA, NA,
-    11, 12, NA, NA, NA,
-    16, NA, NA, NA, NA
-  ), nrow = 4, byrow = TRUE)
-  expect_equal(replace_lower_right_with_NA(rect_matrix2),
-    expected_rect2,
-    tol = 1e-6
+test_that("replace_lower_right_with_NA handles one-column matrix", {
+  one_col <- matrix(1:5, ncol = 1)
+  result <- replace_lower_right_with_NA(one_col)
+  expect_identical(result, one_col)
+})
+
+test_that("replace_lower_right_with_NA does not modify the original matrix", {
+  original <- matrix(1:9, nrow = 3)
+  original_copy <- original
+  result <- replace_lower_right_with_NA(original)
+  expect_identical(original, original_copy)
+  expect_false(identical(result, original))
+})
+
+test_that("replace_lower_right_with_NA handles ragged structure with integer", {
+  test_matrix <- matrix(
+    c(
+      1, 3, 5, 7, 9,
+      4, 7, 8, 0, 12,
+      9, 10, 0, 0, 15,
+      3, 0, 0, 0, 0,
+      6, 2, 0, 0, 0
+    ),
+    nrow = 5,
+    byrow = TRUE
   )
 
-  # Test case 4: 1x1 matrix
-  single_cell <- matrix(1, nrow = 1, ncol = 1)
-  expect_equal(replace_lower_right_with_NA(single_cell),
-    single_cell,
-    tol = 1e-6
+  # Test with structure = 2
+  expected_ragged <- matrix(
+    c(
+      1, 3, 5, 7, 9,
+      4, 7, 8, 0, 12,
+      9, 10, 0, 0, 15,
+      3, 0, 0, 0, NA,
+      6, 2, NA, NA, NA
+    ),
+    nrow = 5,
+    byrow = TRUE
```

Comment:
```suggestion
  expected_ragged <- matrix(
    c(
      1, 3, 5, 7, 9, 10,
      4, 7, 8, 0, 12, 7,
      9, 10, 0, 0, 15, 10,
      3, 0, 0, 0, NA, NA,
      6, 2, NA, NA, NA, NA
    ),
    nrow = 5,
    byrow = TRUE
```

Reply:
I know this will work but slight preference that this example has the same structure throughout (rather than one NA on the second to last row

Reply:
I'm not sure I understand sorry? Don't we want to test different structures and sizes?

Reply:
Yeah I guess its fine -- I know you want to test the random jaggedness but I also think the example makes more sense and is more realistic when the jaggedness is consistent (e.g. `structure = c(2, 2, 2)`)

Reply:
But I don't feel strongly so probably fine as is.